server/auth.py

```diff
@@ -0,0 +1,52 @@
+import os
+from base64 import b64encode
+
+import jwt
+from jwt import InvalidTokenError
+
+from .config import API_JWT_PUBLIC_KEY
+from .core import Module
+from .decorators import with_logger
+from .exceptions import AuthenticationError
+
+
+def is_key(s: str):
+    if s.startswith("-----BEGIN") or s.startswith("ssh-rsa"):
+        return True
+    return False
+
+
+@with_logger
+class AuthModule(Module):
+    def __init__(self) -> None:
+        super(AuthModule, self).__init__()
+
+        self.pub_key = API_JWT_PUBLIC_KEY
+
+        if API_JWT_PUBLIC_KEY and not is_key(API_JWT_PUBLIC_KEY):  # pragma: no cover
+            with open(API_JWT_PUBLIC_KEY) as f:
+                self.pub_key = f.read()
+
+
+mod = AuthModule()
+
+if mod.pub_key:
+    @mod.route("auth")
+    async def handle_auth(conn, message):
+        assert mod.pub_key
+        token = message["token"]
+        try:
+            token = jwt.decode(token, mod.pub_key, algorithms="RS256")
+            user_id = token["user_id"]
+        except (InvalidTokenError, KeyError):
+            raise AuthenticationError("Token signature was invalid")
+
+        new_irc_password = b64encode(os.urandom(30)).decode()
+        conn.send({
+            "command": "irc_password",
+            "password": new_irc_password
```

Comment:
Correct me if I am wrong, but I think the way it works is. The first time somebody logs into IRC the username is associated with a password in our case this is 
![grafik](https://user-images.githubusercontent.com/19996713/71360984-e0c95d00-2591-11ea-8387-7f3d89f6f2a5.png)
Something pretty strange. And IRC will only let u log in if the second time it is still the same password. This way you can reserve your username. Now u send a new IRC password everytime the user logs in, so this won't work(if I got it right). So we either generate a random password per user and it stays the same or maybe the client saves it the first time he logs in to this has issues when he changes clients. Or like we do now we need to derive it from the password, which is bad because the sever is not supposed to know that. Probably storing a random password in db and handing that out on login would be best. @Brutus5000 correct me if I am wrong and you as well @Askaholic

Reply:
https://github.com/FAForever/docker-unrealircd
Yes i think i am right

Reply:
So way to go if u ask me is make a new column and store a random string there per user

Reply:
Also I think we might just do that in the java-api. We can make a database migration and add the column and fill it with a random string then just add the field in the java-api user entity... Then the client can ask the api for the irc password

Reply:
The server updates each user's IRC password every time they log in. I guess this is in case the user changes their password. It would probably make the most sense for the API to handle this entirely, and only update the password when the user actually changes their password.